<a href="https://colab.research.google.com/github/SharonValdez/Prediction-of-Product-Sales/blob/main/Prediction_of_Product_Sales_SharonValdez.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Prediction of Product Sales
 - Author: Sharon Valdez
___

## Project Overview

## Load and Inspect Data

In [1]:
# Import Pandas
import pandas as pd

In [2]:
# Load the data set
fpath = "/content/drive/MyDrive/CodingDojo/01-Fundamentals/Week02/Data/sales_predictions_2023.csv"
df = pd.read_csv(fpath)

In [3]:
# Obtain summary info
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8523 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8523 non-null   object 
 1   Item_Weight                7060 non-null   float64
 2   Item_Fat_Content           8523 non-null   object 
 3   Item_Visibility            8523 non-null   float64
 4   Item_Type                  8523 non-null   object 
 5   Item_MRP                   8523 non-null   float64
 6   Outlet_Identifier          8523 non-null   object 
 7   Outlet_Establishment_Year  8523 non-null   int64  
 8   Outlet_Size                6113 non-null   object 
 9   Outlet_Location_Type       8523 non-null   object 
 10  Outlet_Type                8523 non-null   object 
 11  Item_Outlet_Sales          8523 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 799.2+ KB


In [4]:
# Obtain the first five rows
df.head()

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
0           FDA15         9.30          Low Fat         0.016047   
1           DRC01         5.92          Regular         0.019278   
2           FDN15        17.50          Low Fat         0.016760   
3           FDX07        19.20          Regular         0.000000   
4           NCD19         8.93          Low Fat         0.000000   

               Item_Type  Item_MRP Outlet_Identifier  \
0                  Dairy  249.8092            OUT049   
1            Soft Drinks   48.2692            OUT018   
2                   Meat  141.6180            OUT049   
3  Fruits and Vegetables  182.0950            OUT010   
4              Household   53.8614            OUT013   

   Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
0                       1999      Medium               Tier 1   
1                       2009      Medium               Tier 3   
2                       1999      Medium               Tier 1   
3                       1998         NaN               Tier 3   
4                       1987        High               Tier 3   

         Outlet_Type  Item_Outlet_Sales  
0  Supermarket Type1          3735.1380  
1  Supermarket Type2           443.4228  
2  Supermarket Type1          2097.2700  
3      Grocery Store           732.3800  
4  Supermarket Type1           994.7052

## Clean Data

**1. How many rows and columns?**

In [5]:
# Show the number of rows and columns using .shape
df.shape

(8523, 12)

rows: 8523


columns: 12

**2. What are the datatypes of each variable?**

From the summary info above - the data type of each column is:

Item_Identifier is an object.

Item_Weight is a float.

Item_Fat_Content is an object.

Item_Visibility is a float.

Item_Type is an object.

Item_MRP is a float.

Outlet_Identifier is an object.

Outlet_Establishment_Year is an integer.

Outlet_Size is an object.

Outlet_Location_Type is an object.

Outlet_Type is an object.

Item_Outlet_Sales is a float.

**3. Are there duplicates? If so, drop any duplicates.**

In [6]:
# Identify any duplicate rows.
# Create a filter where duplicates are labeled as True
duplicated_rows = df.duplicated()
# Count the duplicates
df.duplicated().sum()

0

There are no duplicates in this data set.

**4. Identify missing values**

From the inspection above - there are two columns with missing values: Item_Weight and Outlet_Size.

**5) Address the missing values by using a placeholder value.**

The Item_Weight column is a float type, so I will use a number to fill the missing values.

In [7]:
# Show the null values for the item weight column.
df['Item_Weight'].isna()

0       False
1       False
2       False
3       False
4       False
        ...  
8518    False
8519    False
8520    False
8521    False
8522    False
Name: Item_Weight, Length: 8523, dtype: bool

In [8]:
# Add up the true statements and save as a variable
null_weights = df['Item_Weight'].isna().sum()
null_weights

1463

In [9]:
# Obtain summary statistics for the Item_Weight column
df['Item_Weight'].describe()

count    7060.000000
mean       12.857645
std         4.643456
min         4.555000
25%         8.773750
50%        12.600000
75%        16.850000
max        21.350000
Name: Item_Weight, dtype: float64

In [10]:
# Fill the missing values with the average item weight
df['Item_Weight'] = df['Item_Weight'].fillna(12.86)

The Outlet_Size column is an object type, so I will use MISSING to fill the missing values.

In [11]:
# Show the null values for the outlet size column.
df['Outlet_Size'].isna()

0       False
1       False
2       False
3        True
4       False
        ...  
8518    False
8519     True
8520    False
8521    False
8522    False
Name: Outlet_Size, Length: 8523, dtype: bool

In [12]:
# Add up the true statements and save as a variable
df['Outlet_Size'].isna().sum()

2410

In [13]:
# Use the MISSING placeholder to fill the null values
df['Outlet_Size'] = df['Outlet_Size'].fillna('MISSING')

**6) Confirm that there are no missing values after addressing them.**

In [14]:
# Verify the number of null values
df.isna().sum()

Item_Identifier              0
Item_Weight                  0
Item_Fat_Content             0
Item_Visibility              0
Item_Type                    0
Item_MRP                     0
Outlet_Identifier            0
Outlet_Establishment_Year    0
Outlet_Size                  0
Outlet_Location_Type         0
Outlet_Type                  0
Item_Outlet_Sales            0
dtype: int64

**7) Find and fix any inconsistent categories of data (for example: fix cat, Cat, and cats so that they are consistent).**

In [15]:
# Create a variable to isolate only the string columns
string_cols = df.select_dtypes('object').columns
string_cols

Index(['Item_Identifier', 'Item_Fat_Content', 'Item_Type', 'Outlet_Identifier',
       'Outlet_Size', 'Outlet_Location_Type', 'Outlet_Type'],
      dtype='object')

In [16]:
# Obtain the value counts for these string columns
for col in string_cols:
  print(f'Value counts for {col}')
  print(df[col].value_counts())
# Increase readability by adding an empty line
  print('\n')

Value counts for Item_Identifier
FDW13    10
FDG33    10
NCY18     9
FDD38     9
DRE49     9
         ..
FDY43     1
FDQ60     1
FDO33     1
DRF48     1
FDC23     1
Name: Item_Identifier, Length: 1559, dtype: int64


Value counts for Item_Fat_Content
Low Fat    5089
Regular    2889
LF          316
reg         117
low fat     112
Name: Item_Fat_Content, dtype: int64


Value counts for Item_Type
Fruits and Vegetables    1232
Snack Foods              1200
Household                 910
Frozen Foods              856
Dairy                     682
Canned                    649
Baking Goods              648
Health and Hygiene        520
Soft Drinks               445
Meat                      425
Breads                    251
Hard Drinks               214
Others                    169
Starchy Foods             148
Breakfast                 110
Seafood                    64
Name: Item_Type, dtype: int64


Value counts for Outlet_Identifier
OUT027    935
OUT013    932
OUT049    930
OUT046    930


There are inconsistent values in the Item_Fat_Content column.

In [17]:
# Isolate the value counts for the Item_Fat_Content column
df['Item_Fat_Content'].value_counts()

Low Fat    5089
Regular    2889
LF          316
reg         117
low fat     112
Name: Item_Fat_Content, dtype: int64

In [18]:
# Standardize values in the Item_Fat_Content column
df['Item_Fat_Content'] = df['Item_Fat_Content'].replace({'LF':'Low Fat', 'reg':'Regular', 'low fat':'Low Fat'})

In [19]:
# Verify the change by checking value counts
df['Item_Fat_Content'].value_counts()

Low Fat    5517
Regular    3006
Name: Item_Fat_Content, dtype: int64

**8) For any numerical columns, obtain the summary statistics of each (min, max, mean).**

In [20]:
# Create a variable to isolate number columns
num_cols = df.select_dtypes('number').columns
num_cols

Index(['Item_Weight', 'Item_Visibility', 'Item_MRP',
       'Outlet_Establishment_Year', 'Item_Outlet_Sales'],
      dtype='object')

In [21]:
# Obtain the summary of statistics for the number columns
df[num_cols].describe()

Item_Weight  Item_Visibility     Item_MRP  Outlet_Establishment_Year  \
count  8523.000000      8523.000000  8523.000000                8523.000000   
mean     12.858049         0.066132   140.992782                1997.831867   
std       4.226124         0.051598    62.275067                   8.371760   
min       4.555000         0.000000    31.290000                1985.000000   
25%       9.310000         0.026989    93.826500                1987.000000   
50%      12.860000         0.053931   143.012800                1999.000000   
75%      16.000000         0.094585   185.643700                2004.000000   
max      21.350000         0.328391   266.888400                2009.000000   

       Item_Outlet_Sales  
count        8523.000000  
mean         2181.288914  
std          1706.499616  
min            33.290000  
25%           834.247400  
50%          1794.331000  
75%          3101.296400  
max         13086.964800

## Exploratory Data Analysis

## Feature Inspection